ext/TrixiConvexECOSExt.jl

```diff
@@ -0,0 +1,158 @@
+# Package extension for adding Convex-based features to Trixi.jl
+module TrixiConvexECOSExt
+
+# Required for coefficient optimization in P-ERK scheme integrators
+if isdefined(Base, :get_extension)
+    using Convex: MOI, solve!, Variable, minimize, evaluate
+    using ECOS
+else
+    # Until Julia v1.9 is the minimum required version for Trixi.jl, we still support Requires.jl
+    using ..Convex: MOI, solve!, Variable, minimize, evaluate
+    using ..ECOS
+end
```

Comment:
Can we please get an explicit list of stuff that needs to be imported from `ECOS` here?